File: CAP.8/Atv2.py
def fatorial(n: int) -> int:
    f = 1
    for i in range(2, n + 1):
        f *= i
    return f

while True:
    print("1) Quadrado\n2) Fatorial\n3) Sair")
    try:
        opcao = int(input("Escolha: "))
    except ValueError:
        print("Opcao invalida!\n"); continue

    if opcao == 1:
        x = int(input("Numero: "))
        print("Quadrado =", x * x)
    elif opcao == 2:
        n = int(input("Numero (n>=0): "))
        print("Fatorial =", fatorial(n))
    elif opcao == 3:
        print("Saindo...")
        break
    else:
        print("Opcao invalida!")

File: CAP.8/Atv2.c
#include <stdio.h>

long fatorial(int n) { 
    long f = 1; 
    for (int i = 2; i <= n; i++) f *= i; 
    return f; 
}

int main() {
    int opcao;
    do {
        printf("1) Quadrado\n2) Fatorial\n3) Sair\nEscolha: ");
        if (scanf("%d", &opcao) != 1) return 0;

        switch (opcao) {
            case 1: {
                int x; printf("Numero: "); scanf("%d", &x);
                printf("Quadrado = %d\n", x * x);
                break;
            }
            case 2: {
                int n; printf("Numero (n>=0): "); scanf("%d", &n);
                printf("Fatorial = %ld\n", fatorial(n));
                break;
            }
            case 3:
                printf("Saindo...\n");
                break;
            default:
                printf("Opcao invalida!\n");
        }
    } while (opcao != 3);
    return 0;
}
